### PR TITLE
(#3174) Store non-normalized package version

### PR DESCRIPTION
**Pull request by AdmiringWorm**
## Description Of Changes

In a previous change we normalized all versions that was being used
in Chocolatey CLI. This caused certain packages that uses non-normalized
version numbers to start failing to look up additional information in other
products.

To work around this problem we need to store the non-normalized version
of the package when scripts are being ran. This is done by adding a new
environment variable that can be used, this environment variable is considered
private and is not for public consumption.

## Motivation and Context

To be able to look up the proper information for legacy packages.

## Testing

Enable the hermes choco-test-packages source.

1. Run `choco install test-environment --version 0.9`
2. Verify `chocolateyPackageVersion=0.9.0` is outputted.
3. Verify `packageVersion=0.9.0` is outputted.
4. Verify `chocolateyPackageVersionOriginal=0.9` is outputted.
5. Verify `packageVersionOriginal=0.9` is outputted.
6. Run `choco upgrade test-environment`
7. Verify in the before modify script
   - Verify `chocolateyPackageVersion=0.9.0` is outputted.
   - Verify `packageVersion=0.9.0` is outputted.
   - Verify `chocolateyPackageVersionOriginal=0.9` is outputted.
   - Verify `packageVersionOriginal=0.9` is outputted.
8. Verify in the install script
   - Verify `chocolateyPackageVersion=1.0.0` is outputted.
   - Verify `packageVersion=1.0.0` is outputted.
   - Verify `chocolateyPackageVersionOriginal=1.0.0` is outputted.
   - Verify `packageVersionOriginal=1.0.0` is outputted.

Rerun tests with licensed extension installed.

### Operating Systems Testing

- Windows 10
- Windows Server 2022

## Change Types Made

* [x] Bug fix (non-breaking change).
* [x] Feature / Enhancement (non-breaking change).
* [ ] Breaking change (fix or feature that could cause existing functionality to change).
* [ ] Documentation changes.
* [ ] PowerShell code changes.

## Change Checklist

* [ ] Requires a change to the documentation.
* [ ] Documentation has been updated.
* [ ] Tests to cover my changes, have been added.
* [ ] All new and existing tests passed?
* [ ] PowerShell code changes: PowerShell v2 compatibility checked?

## Related Issue

#3174

### CODE REVIEW

File: src/chocolatey/infrastructure.app/services/PowershellService.cs [hunk 1]
```diff
@@ -446,6 +446,10 @@ public void PreparePowerShellEnvironment(IPackageSearchMetadata package, Chocola
             Environment.SetEnvironmentVariable("packageTitle", package.Title);
             Environment.SetEnvironmentVariable("chocolateyPackageVersion", package.Identity.Version.ToNormalizedStringChecked());
             Environment.SetEnvironmentVariable("packageVersion", package.Identity.Version.ToNormalizedStringChecked());
+            // We use ToStringSafe on purpose here. There is a need for the version
+            // the package specified, not the normalized version we want users to use.
+            Environment.SetEnvironmentVariable("chocolateyPackageVersionOriginal", package.Identity.Version.ToStringSafe());
```
Comment (gep13) [inline src/chocolatey/infrastructure.app/services/PowershellService.cs:451]: @pauby before moving forward with this PR, what are your thoughts on this variable name? `chocolateyPackageVersionOriginal` This is the potentially non-normalized package version, i.e. 1.2, where as the `chocolateyPackageVersion` variable would contain the normalized version, i.e. 1.2.0.

Comment (pauby) [reply src/chocolatey/infrastructure.app/services/PowershellService.cs:451]: What about `chocolateyPackageVersionPushed`? `Original` doesn't feel right to me and I think will cause confusion.

Comment (gep13) [reply src/chocolatey/infrastructure.app/services/PowershellService.cs:451]: Not sure `Pushed` is right here either. This environment variable doesn't really have anything to do with pushing to package, but rather what was the original version number before it was normalised, I.e. what is the original package version in the nuspec file for the package.

Comment (pauby) [reply src/chocolatey/infrastructure.app/services/PowershellService.cs:451]: > what is the original package version in the nuspec file for the package.

Is that not the version number it was given when it was pushed?

Comment (gep13) [reply src/chocolatey/infrastructure.app/services/PowershellService.cs:451]: When it was `Packed` perhaps, but nothing really to do with when it was `Pushed`.

Comment (gep13) [reply src/chocolatey/infrastructure.app/services/PowershellService.cs:451]: @pauby just thinking about this again... what about `chocolateyPackageVersionFromNuSpec` and `packageVersionFromNuSpec`?

Comment (pauby) [reply src/chocolatey/infrastructure.app/services/PowershellService.cs:451]: I like those. Lets use them.

Comment (gep13) [inline src/chocolatey/infrastructure.app/services/PowershellService.cs:451]:
```suggestion
            Environment.SetEnvironmentVariable("chocolateyPackageVersionFromNuSpec", package.Identity.Version.ToStringSafe());

```

File: src/chocolatey/infrastructure.app/services/PowershellService.cs [hunk 2]
```diff
@@ -446,6 +446,10 @@ public void PreparePowerShellEnvironment(IPackageSearchMetadata package, Chocola
             Environment.SetEnvironmentVariable("packageTitle", package.Title);
             Environment.SetEnvironmentVariable("chocolateyPackageVersion", package.Identity.Version.ToNormalizedStringChecked());
             Environment.SetEnvironmentVariable("packageVersion", package.Identity.Version.ToNormalizedStringChecked());
+            // We use ToStringSafe on purpose here. There is a need for the version
+            // the package specified, not the normalized version we want users to use.
+            Environment.SetEnvironmentVariable("chocolateyPackageVersionOriginal", package.Identity.Version.ToStringSafe());
+            Environment.SetEnvironmentVariable("packageVersionOriginal", package.Identity.Version.ToStringSafe());
```
Comment (gep13) [inline src/chocolatey/infrastructure.app/services/PowershellService.cs:452]:
```suggestion
            Environment.SetEnvironmentVariable("packageVersionFromNuSpec", package.Identity.Version.ToStringSafe());

```

File: tests/chocolatey-tests/commands/choco-install.Tests.ps1 [hunk 3]
```diff
@@ -1863,6 +1863,39 @@ To install a local, or remote file, you may use:
...
+        It 'Outputs <Name> as <Value>' -ForEach @(@{
+            Name = 'chocolateyPackageVersion'
+            Value= '0.9.0'
+        }
+        @{
+            Name = 'packageVersion'
+            Value= '0.9.0'
+        }
+        @{
+            Name = 'chocolateyPackageVersionOriginal'
```
Comment (gep13) [inline tests/chocolatey-tests/commands/choco-install.Tests.ps1:1888]:
```suggestion
            Name = 'chocolateyPackageVersionFromNuSpec'

```

File: tests/chocolatey-tests/commands/choco-install.Tests.ps1 [hunk 4]
```diff
@@ -1863,6 +1863,39 @@ To install a local, or remote file, you may use:
...
+            Value= '0.9'
+        }
+        @{
+            Name = 'packageVersionOriginal'
```
Comment (gep13) [inline tests/chocolatey-tests/commands/choco-install.Tests.ps1:1892]:
```suggestion
            Name = 'packageVersionFromNuSpec'

```

**Review by gep13 — CHANGES_REQUESTED** [review]

File: src/chocolatey/infrastructure.app/services/PowershellService.cs (hunk 1 above)
Comment (gep13) [reply src/chocolatey/infrastructure.app/services/PowershellService.cs:451]: @pauby just had a chat with Kim, and he has made a better suggestion, how about `chocolateyNuSpecVersion` and `nuSpecVersion`?

Comment (pauby) [reply src/chocolatey/infrastructure.app/services/PowershellService.cs:451]: I would prefer `chocolateyPackageNuspecVersion` as it refers to something specific for the package. I can see us publicly documenting this in future, and the `chocolateyPackage` part is consistent with package specific environment variables.

Comment (AdmiringWorm) [reply src/chocolatey/infrastructure.app/services/PowershellService.cs:451]: After thinking about it and going back and forth on it, I'll get the PR updated to use `chocolateyPackageNuspecVersion`.

**Review by gep13 — APPROVED** [review]
LGTM!

**Comment by gep13** [comment]
@AdmiringWorm thanks for getting this fixed up!

**Pull request merged**